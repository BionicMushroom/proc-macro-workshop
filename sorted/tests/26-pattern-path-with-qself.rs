trait TestTrait {
    const VALUE: u32;
}

struct TestStruct;

impl TestTrait for TestStruct {
    const VALUE: u32 = 0;
}

#[sorted::check]
fn main() {
    let x = 0;

    #[sorted]
    match x {
        <TestStruct as TestTrait>::VALUE => println!("first"),
        _ => println!("other"),
    }
}
