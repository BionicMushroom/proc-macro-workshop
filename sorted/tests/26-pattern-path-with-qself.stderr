error: qualified paths are unsupported by #[sorted]
  --> tests/26-pattern-path-with-qself.rs:17:9
   |
17 |         <TestStruct as TestTrait>::VALUE => println!("first"),
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
